### PR TITLE
add a tutorial doc

### PR DESCRIPTION
**Pull request by eklavya**
Apologies for putting it in root. Let me know where it should go.

### CODE REVIEW

File: Rho.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+val api = new RhoService[IO] {
+    "Description of api endpoint" **
+    GET / "somePath" / pathVar[Int]("someInt", "parameter description") +? paramD[String]("name", "parameter description") |>> { (someInt: Int, name: String) => {
+            Ok("result")
+        }
+    }
+}
+```
+
+So we start by creating a new RhoService which will be converted to an HttpService later. This RhoService will contain our api endpoints. Every api endpoint starts with a description for that endpoint and then the implementation. We describe the HTTP method (GET/POST/DELETE etc) and then a description of the url.
```
Comment (bryce-anderson) [inline Rho.md]: It doesn't _have_ to start with a description if it doesn't want to.

File: Rho.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        )
+      ),
+      example =
+        """{"name" : "MyClass1",  "category" : "A", "tags" : ["tag1", "tag2"], "someBool" : false, "someObj": {}}""".some
+    )
+  )
+
+// register this model
+SwaggerSupport[IO].createRhoMiddleware(
+              swaggerFormats = DefaultSwaggerFormats
```
Comment (jcranky) [inline Rho.md]: I'm confused with this indentation. Maybe 104 and 105 should be in a single line?

Comment (eklavya) [reply Rho.md]: Do you still feel that with the changes, this was scalafmt btw.

Comment (jcranky) [reply Rho.md]: Yes, but if it was scalafmt, maybe it is not worth fixing :(

File: README.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ val httpService = new RhoService {
 }
 ```
 
-See the [wiki](https://github.com/http4s/rho/wiki) and the [tests](https://github.com/http4s/rho/blob/master/core/src/test/scala/ApiExamples.scala) for more examples.
+See the [tutorial](https://github.com/eklavya/rho/blob/master/Rho.md), [wiki](https://github.com/http4s/rho/wiki) and the [tests](https://github.com/http4s/rho/blob/master/core/src/test/scala/ApiExamples.scala) for more examples.
```
Comment (jcranky) [inline README.md]: Should this point to your fork?

Comment (ChristopherDavenport) [reply README.md]: 👍

Comment (ChristopherDavenport) [reply README.md]: Easy thing to have happen by accident. 👍

Comment (eklavya) [reply README.md]: yup, just me being silly :)

**Pull request merged**